Lib/asyncio/unix_events.py

```diff
@@ -77,6 +76,7 @@ def close(self):
                               ResourceWarning,
                               source=self)
                 self._signal_handlers.clear()
+        super().close()
```

Comment:
Otherwise, an active signal wakeup file descriptor may be scrapped too early.

Reply:
At the very least this deserves to be a comment in the source code. IIUC `super().close()` is `selector_events.BaseSelectorEventLoop.close()`, which closes the socketpair, and that would break the check you've added to `remove_signal_handler()`. But I'm not sure I like that check, see below.

Reply:
Comment added in https://github.com/python/cpython/pull/96807/commits/3b4975a4241964fa28f5d7398604b36ba3b85d3d.